paddle/operators/sequence_concat_op.h

```diff
@@ -0,0 +1,167 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+#include "paddle/framework/op_registry.h"
+#include "paddle/operators/strided_memcpy.h"
+
+namespace paddle {
+namespace operators {
+
+using Tensor = framework::Tensor;
+using LoDTensor = framework::LoDTensor;
+using LoD = framework::LoD;
+
+// Concat LoD, the initialized LoD of Output is lod(x0),
+// if axis is not 0, the LoD(Out) will be the same as Inputs, if axis is 0:
+// Case1:
+//  There is one level, the Output LoD will be modified:
+//  LoD(x0) = {{0,2,4}}
+//  LoD(x1) = {{0,1,5}}
+//  LoD(Out) = {{0,3,9}}
+// Case2:
+//  There is two level, and concat level is 1,
+//  the Output LoD will be modified as followed:
+//  LoD(x0) = {{0,2,4}, {0,1,2,3,4}}
+//  LoD(x1) = {{0,3,5}, {0,1,3,4,5}}
+//  LoD(Out) = {{0,5,9}, {0,1,2,4,5,6,7,8,9}}
+template <typename T>
+LoD concatLoD(const std::vector<const T*> ins, const size_t axis,
+              const size_t level) {
+  auto out_lod = ins[0]->lod();
+  const size_t n = ins.size();
+  if (axis == 0UL) {
+    if (level == 0) {
+      for (size_t i = 1; i < n; ++i) {
+        for (size_t j = 0; j < ins[i]->lod()[0].size(); ++j) {
+          out_lod[0][j] += ins[i]->lod()[0][j];
+        }
+      }
+    } else if (level == 1) {
+      PADDLE_ENFORCE_EQ(ins[0]->NumLevels(), 2UL,
+                        "If the level is 1, all of the inputs "
+                        "should be the the nested sequence.");
```

Comment:
- `the the` --> an extra `the`
- nested sequence --> nested sequences.

Reply:
Done.